File: 3.3_Air Quality Monitor.cpp
#include <iostream>
#include <cstdlib>
#include <ctime>

class AirQualityMonitor {
private:
    float pm25_sensor;
    float gas_sensor;
public:
    AirQualityMonitor() : pm25_sensor(0), gas_sensor(0) {}

    void measurePM25() {
        // 模拟PM2.5传感器测量，范围在20附近波动
        pm25_sensor = 20 + static_cast<float>(rand()) / (static_cast<float>(RAND_MAX / 10));
    }

    void measureGas() {
        // 模拟气体传感器测量，范围在300附近波动
        gas_sensor = 300 + static_cast<float>(rand()) / (static_cast<float>(RAND_MAX / 100));
    }

    void displayResult() {
        // 显示测量结果
        std::cout << "PM2.5 Level: " << pm25_sensor << std::endl;
        std::cout << "Gas Concentration: " << gas_sensor << std::endl;
    }
};

// 示例用法
int main() {
    srand(static_cast<unsigned>(time(nullptr)));

    AirQualityMonitor monitor;
    monitor.measurePM25();
    monitor.measureGas();
    monitor.displayResult();
    return 0;
}

File: 3.3_Air Quality Monitor.py
# oding = utf-8
# -*- coding:utf-8 -*-
import random

class AirQualityMonitor:
    def __init__(self):
        # 初始化传感器读数
        self.pm25_sensor = 0
        self.gas_sensor = 0

    def measure_pm25(self):
        # 模拟PM2.5传感器测量，范围在20附近波动
        self.pm25_sensor = 20 + random.uniform(-5, 5)

    def measure_gas(self):
        # 模拟气体传感器测量，范围在300附近波动
        self.gas_sensor = 300 + random.uniform(-50, 50)

    def display_result(self):
        # 显示测量结果
        print(f"PM2.5 Level: {self.pm25_sensor}")
        print(f"Gas Concentration: {self.gas_sensor}")

# 示例用法
if __name__ == "__main__":
    monitor = AirQualityMonitor()
    monitor.measure_pm25()
    monitor.measure_gas()
    monitor.display_result()